ico/tests/contracts/test_security_token.py

```diff
@@ -113,15 +130,120 @@ def zero_address() -> str:
     return "0x0000000000000000000000000000000000000000"
 
 
+@pytest.fixture
+def voting_contract(chain, team_multisig, mock_kyc, security_token, security_token_verifier) -> Contract:
+    """Create the Voting Contract."""
+
+    # CheckpointToken _token, MockKYC _KYC, bytes32 name, bytes32 URI, uint256 _type, uint256 _hash, bytes32[] _options
+    args = [
+        security_token.address,
+        mock_kyc.address,
+        to_bytes(text="Voting X"),
+        to_bytes(text="http://tokenmarket.net"),
+        123,
+        0,
+        [to_bytes(text="Vested for voting")]
+    ]
+
+    tx = {
+        "from": team_multisig
+    }
+
+    contract, hash_ = chain.provider.deploy_contract('VotingContract', deploy_args=args, deploy_transaction=tx)
+
+    check_gas(chain, hash_)
+
+    return contract
+
+
+@pytest.fixture
+def payout_contract(chain, team_multisig, mock_kyc, security_token, test_token, security_token_verifier) -> Contract:
+    """Create the Voting Contract."""
+
+    # CheckpointToken _token, MockKYC _KYC, bytes32 name, bytes32 URI, uint256 _type, uint256 _hash, bytes32[] _options
+    # address, address, address, bytes32, bytes32, uint256, uint256, bytes32[]
+    args = [
+        security_token.address,
+        test_token.address,
+        mock_kyc.address,
+        to_bytes(text="Pay X"),
+        to_bytes(text="http://tokenmarket.net"),
+        123,
+        0,
+        [to_bytes(text="Vested for dividend")]
+    ]
+
+    tx = {
+        "from": team_multisig
+    }
+
+    contract, hash_ = chain.provider.deploy_contract('PayoutContract', deploy_args=args, deploy_transaction=tx)
+
+    check_gas(chain, hash_)
+
+    return contract
+
 #
 # ERC-20 fixtures
 #
 
 @pytest.fixture
-def security_token(chain, team_multisig, security_token_name, security_token_symbol, security_token_initial_supply) -> Contract:
+def test_token(chain, team_multisig, token_name, token_symbol, security_token_initial_supply, release_agent, customer) -> Contract:
+    """Create a Crowdsale token where transfer restrictions have been lifted."""
+
+    args = [token_name, token_symbol, security_token_initial_supply, 18, True]  # Owner set
+
+    tx = {
+        "from": team_multisig
+    }
+
+    token, hash = chain.provider.deploy_contract('CrowdsaleToken', deploy_args=args, deploy_transaction=tx)
+
+    token.transact({"from": team_multisig}).setReleaseAgent(team_multisig)
+    token.transact({"from": team_multisig}).releaseTokenTransfer()
+
+    return token
+
+
+@pytest.fixture
+def security_token_verifier(chain, team_multisig) -> Contract:
+    """Create the transaction verifier contract."""
+
+    tx = {
+        "from": team_multisig
+    }
+
+    contract, hash_ = chain.provider.deploy_contract('MockSecurityTransferAgent', deploy_transaction=tx)
+
+    check_gas(chain, hash_)
+
+    return contract
+
+
+@pytest.fixture
+def mock_kyc(chain, team_multisig, customer) -> Contract:
```

Comment:
Now that we have replaced MockKYC with `KYCInterface`, should we use `BasicKYC` instead? Not very important though. But why mock when we already have an actual smart contract?

Reply:
Ok, my bad. I see that it actually deploys `BasicKYC`. However, this fixture needs to be renamed.

Reply:
This is taken care of in feat/test-improvements :slightly_smiling_face: